lib/matplotlib/axes/_axes.py

```diff
@@ -78,6 +78,24 @@ def _plot_args_replacer(args, data):
                          "multiple plotting calls instead.")
 
 
+class _inset_locator(object):
```

Comment:
would just make this a function
```
def _make_inset_locator(rect, trans, parent):
    rect = ...

    def inset_locator(ax, renderer):
        ...
        return bb

    return inset_locator
```
Avoids having to manipulate state yourself.

Not a big deal though.

Reply:
I didn't know the scope would be OK if you did that...